core/src/main/java/org/geysermc/geyser/translator/protocol/java/inventory/JavaOpenScreenTranslator.java

```diff
@@ -46,6 +48,14 @@ public void translate(GeyserSession session, ClientboundOpenScreenPacket packet)
 
         InventoryTranslator newTranslator = InventoryTranslator.inventoryTranslator(packet.getType());
         Inventory openInventory = session.getOpenInventory();
+
+        // Hack: ViaVersion translates the old (pre 1.20) smithing table to a furnace (does not work for Bedrock). We can detect this and translate it back to a smithing table.
+        if (session.isOldSmithingTable() &&
+            packet.getType() == ContainerType.FURNACE &&
+            MessageTranslator.getTranslationKey(packet.getTitle()).equals("container.upgrade")) {
```

Comment:
This seems redundant and also assuming a Component is a TranslatableComponent is a Very Bad Idea. I would make a static final `Component.translatable("container.upgrade")` and compare the title to that.

Reply:
Valid point, i cant even remember when i put that equals check into the translation key thing.. my bad